src/main/java/uk/nhs/tis/trainee/notifications/event/ProgrammeMembershipListener.java

```diff
@@ -78,8 +78,8 @@ public void handleProgrammeMembershipDelete(ProgrammeMembershipEvent event)
       throws SchedulerException {
     log.info("Handling programme membership delete event {}.", event);
     if (event.recrd() != null && event.recrd().getData() != null) {
-      ProgrammeMembership programmeMembership = new ProgrammeMembership();
-      programmeMembership.setTisId(event.tisId()); //delete messages have empty recrd data
+      ProgrammeMembership programmeMembership = mapper.toEntity(event.recrd().getData());
+      programmeMembership.setTisId(event.tisId());
       programmeMembershipService.deleteNotificationsFromScheduler(programmeMembership);
       programmeMembershipService.deleteScheduledNotificationsFromDb(programmeMembership);
```

Comment:
This uses the personId to find the notifications; the existing code would not have worked because personId would be null.